config/confighttp/confighttp.go

```diff
@@ -32,6 +33,28 @@ import (
 	"go.opentelemetry.io/collector/internal/middleware"
 )
 
+type CompressionType string
+
+const (
+	CompressionUnsupported CompressionType = ""
+	CompressionGzip        CompressionType = "gzip"
+	CompressionZlib        CompressionType = "zlib"
+	CompressionDeflate     CompressionType = "deflate"
+	CompressionSnappy      CompressionType = "snappy"
+	CompressionZstd        CompressionType = "zstd"
+)
+
+var (
+	// HTTPCompressionKeyMap is the map of OpenTelemetry compression types to HTTP registered compression types.
+	HTTPCompressionKeyMap = map[CompressionType]string{
+		CompressionGzip:    "gzip",
+		CompressionZlib:    "zlib",
+		CompressionDeflate: "deflate",
+		CompressionSnappy:  "snappy",
+		CompressionZstd:    "zstd",
+	}
```

Comment:
This isn't necessary, is it?  Since the underlying type of `CompressionType` is `string` it should be possible to use a `CompressionType` value as a string.  If it can't be used directly then it can be coerced with `string(val)`.  It looks like this map is also used for validation, but that can be done with a set:

```go
validCompressionKeys = map[CompressionType]struct{}{
  CompressionGzip: {},
...
}
```

Then for validation just check to see if the entry is in the set:

```go
_, valid := validCompressionKeys[CompressionType(strVal)]
```

Reply:
done